enterprise/aibridgeproxyd/aibridgeproxyd.go

```diff
@@ -111,3 +134,87 @@ func loadMitmCertificate(certFile, keyFile string) error {
 
 	return nil
 }
+
+// extractPort extracts the port from a host:port string.
+// Returns "443" as the default if no port is specified (standard HTTPS).
+func extractPort(host string) string {
+	if i := strings.LastIndex(host, ":"); i != -1 {
+		return host[i+1:]
+	}
+	return "443"
+}
+
+// handleConnect handles CONNECT requests for HTTPS proxying.
+// It extracts the Coder session token from the Proxy-Authorization header
+// and stores it in ctx.UserData for use by downstream request handlers.
+// Requests without valid credentials are rejected.
+//
+// Clients provide credentials by setting their HTTP Proxy as:
+//
+//	HTTPS_PROXY=http://ignored:<coder-token>@host:port
+//
+// The token is extracted from the password field of basic auth.
+func (s *Server) handleConnect(host string, ctx *goproxy.ProxyCtx) (*goproxy.ConnectAction, string) {
+	// Restrict CONNECT to standard HTTP/HTTPS ports to prevent request smuggling.
+	// Attackers could use non-standard ports to bypass security controls.
+	port := extractPort(host)
+	if port != "443" && port != "80" {
+		s.logger.Warn(s.ctx, "connect to non-standard port",
+			slog.F("host", host),
+			slog.F("port", port),
+		)
+		// return goproxy.RejectConnect, host
```

Comment:
Note: this port restriction makes it harder to test, as the test uses `httptest.NewTLSServer`, which picks a random port, but this rejects anything that's not port 80 or 443. Should we apply this restriction check only to known AI provider hosts?